File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashMap <Integer, Boolean> map = new HashMap<>();
        int index = 0;
        for(int i =0;i<nums.length;i++){
            if(!map.containsKey(nums[i])){
                map.put(nums[i],true);
                nums[index]=nums[i];
                index++;
            }
        }
        return index;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        unique = []
        for num in nums:
            if len(unique) == 0:
                unique.append(num)
            elif num != unique[-1]:
                unique.append(num) # since it is sorted so duplicates are aaju baju so the one going into the list and the last went is same if duplicate
        for i in range(len(unique)):
            nums[i] = unique[i]

        return len(unique)